templates/xcassets/swift3.stencil

```diff
@@ -88,6 +99,40 @@
 
 // MARK: - Implementation Details
 
+{{accessModifier}} struct {{arResourceGroupType}} {
+  {{accessModifier}} fileprivate(set) var name: String
+
+  #if os(iOS) && swift(>=3.2)
+  @available(iOS 11.3, *)
+  {{accessModifier}} var referenceImages: Set<ARReferenceImage> {
+    return ARReferenceImage.referenceImages(in asset: self)
+  }
+
+  @available(iOS 12.0, *)
+  {{accessModifier}} var referenceObjects: Set<ARReferenceObject> {
+    return ARReferenceObject.referenceObjects(in asset: self)
+  }
+  #endif
+}
+
+#if os(iOS) && swift(>=3.2)
+@available(iOS 11.3, *)
+{{accessModifier}} extension ARReferenceImage {
+  static func referenceImages(in asset: {{arResourceGroupType}}) -> Set<ARReferenceImage> {
+    let bundle = Bundle(for: BundleToken.self)
+    return referenceImages(inGroupNamed: asset.name, bundle: bundle) ?? Set()
+  }
+}
+
+@available(iOS 12.0, *)
+{{accessModifier}} extension ARReferenceObject {
+  static func referenceObjects(in asset: {{arResourceGroupType}}) -> Set<ARReferenceObject> {
+    let bundle = Bundle(for: BundleToken.self)
+    return referenceObjects(inGroupNamed: asset.name, bundle: bundle) ?? Set()
+  }
+}
+#endif
```

Comment:
Should only be present if at least one resource of this type exists in the asset catalog, otherwise it quite pollutes the implementation details if no generated constant use it, and it imports (and links to) ARKit for nothing

Reply:
Ehm, if we do it for this, do we do it consistently for the other types? And what if we at some point want to separate the implementation details? Then we need all the implementations.

Reply:
Of course if we start that (which was less necessary before since the other types didn't require to import a less common Framework and we didn't have that many different resource types to begin with, but now that we're growing and spring more resource types…) we might want to apply that concept (only generating the implementation details if there's at least one resource of said type) to all resource types (images, colors, …) ; ARResourceGroups are just the most significant ones for which we'd want that to avoid importing a big framework for nothing, but better make it consistent for all resource types while we're at it 😅

Reply:
I'm not sure I see how separating the implementation details in their own file like we discussed in another issue would be a problem: the templates generating the implementation details could use the context containing all the resource to know what to generate. And a param to force generating each type if people need customization (like if they analyze multiple folders as different inputs in their config file). We'll always find a solution by then when we'll start working on that issue I think, I'm not too worried about that